### ISSUE TITLE
HorizontalPodAutoscaler Does Not Down-Scale, Despite Under-Utilized CPU

### ISSUE THREAD

**Issue by speedplane** [kubernetes/kubernetes#45525]
My HoriozontalPodAutoscaler does not seem to be scaling, even though my pods are way underutilized. This is how I defined my autoscaler:
```
apiVersion: autoscaling/v1
kind: HorizontalPodAutoscaler
metadata:
  name: ocr-extra-scaler
  namespace: ocr-da-cluster
spec:
  scaleTargetRef:
    apiVersion: extensions/v1beta1
    kind: Deployment
    name: ocr-extra
  minReplicas: 1
  maxReplicas: 12
  # We leave a bit of headroom for other requests.
  targetCPUUtilizationPercentage: 80
```

As shown in the dashboard below, my pods are all far underutilized, yet the autoscaler is not downscaling any of them:

![image](https://cloud.githubusercontent.com/assets/671052/25836115/51ed106e-3452-11e7-98d8-3e4e8649b0e1.png)

Here is the output of `kubectl describe hpa`:
```
Name:                                                   ocr-extra-scaler
Namespace:                                              ocr-da-cluster
Labels:                                                 <none>
Annotations:                                            kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"autoscaling/v1","kind":"HorizontalPodAutoscaler","metadata":{"annotations":{},"name":"ocr-extra-scaler","namespace":"ocr-da-cluster"},"s...
CreationTimestamp:                                      Fri, 24 Mar 2017 14:07:06 -0400
Reference:                                              Deployment/ocr-extra
Metrics:                                                ( current / target )
  resource cpu on pods  (as a percentage of request):   0% (0) / 80%
Min replicas:                                           1
Max replicas:                                           12
Events:                                                 <none>
```

I'm not sure if it matters, but I have a bunch of old unused replica sets lying around:
```
>kubectl get rs
NAME                   DESIRED   CURRENT   READY     AGE
ocr-extra-1074021295   0         0         0         40d
ocr-extra-180895804    0         0         0         40d
ocr-extra-2106695714   0         0         0         5d
ocr-extra-2311536872   0         0         0         40d
ocr-extra-269689544    0         0         0         5d
ocr-extra-2893487091   0         0         0         5d
ocr-extra-3012757315   0         0         0         5d
ocr-extra-30228728     0         0         0         40d
ocr-extra-3050980099   0         0         0         40d
ocr-extra-3117826196   0         0         0         40d
ocr-extra-3362659359   0         0         0         5d
ocr-extra-338380943    0         0         0         5d
ocr-extra-3664772186   0         0         0         5d
ocr-extra-3812036703   0         0         0         5d
ocr-extra-3875084349   0         0         0         5d
ocr-extra-4054860708   0         0         0         26d
ocr-extra-4147059556   0         0         0         5d
ocr-extra-4233329942   0         0         0         5d
ocr-extra-490936662    0         0         0         5d
ocr-extra-690099824    12        12        12        4d
ocr-extra-998392029    0         0         0         40d
ocr-main-1556256245    0         0         0         40d
ocr-main-1720347859    0         0         0         40d
ocr-main-2265306178    0         0         0         40d
ocr-main-2381474316    0         0         0         45d
ocr-main-358520578     0         0         0         40d
ocr-main-412667836     0         0         0         40d
ocr-main-690147707     1         1         1         40d
```

**Environment**:
Kubernetes Version:
```
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:33:11Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.6", GitCommit:"114f8911f9597be669a747ab72787e0bd74c9359", GitTreeState:"clean", BuildDate:"2017-03-28T13:36:31Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```
Cloud Environment: Google Cloud Container Engine

**Comment by thomasmodeneis** [comment]
Strage issue, have you tested this on k8s 1.6.6 ?
I have the following cfg and it works like a charm... 

```
apiVersion: autoscaling/v1
kind: HorizontalPodAutoscaler
metadata:
  name: {{ .Chart.Name }}
spec:
  scaleTargetRef:
    kind: Deployment
    name: {{ .Chart.Name }}
  minReplicas: {{ .Values.scaling.minReplicas }}
  maxReplicas: {{ .Values.scaling.maxReplicas }}
  targetCPUUtilizationPercentage: 60
```

Cheers.

**Comment by mattjmcnaughton** [comment]
I'm not sure if this is still a problem for you, but if so, it could be helpful to see the [Horizontal Pod Autoscaler Status Conditions](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/). My understanding is these became available in [1.7](https://github.com/kubernetes/kubernetes/blob/0d6f18287014412450f8810db1d147aa79980fdc/CHANGELOG.md#horizontal-pod-autoscaler).
